python/paddle/distributed/spawn.py

```diff
@@ -202,6 +206,21 @@ def _get_subprocess_env_list(nprocs, options):
                     raise ValueError("The selected xpu card %s cannot found in "
                                      "XPU_VISIBLE_DEVICES (%s)." %
                                      (card_id, ",".join(env_devices_list)))
+    elif get_device() == 'cpu':
+        # TODO check gpu / xpu flag must not exist
+        warnings.warn(
+            "start in CPUONLY mode, because your paddle is cpu version")
```

Comment:
get_device 是判断当前指定的设备，比如 paddle.set_device(paddle.CPUPlace)。也就是说用户编译的Paddle也可能是支持CUDA，只是运行时指定的设备是CPU。

所以这里的报错文案信息要调整下：

`Your model will be trained  under CPUONLY mode by using GLOO, because CPUPlace is specified manually or your installed PaddlePaddle only support CPU Device. `